appgate/resource_appgate_identity_provider_saml.go

```diff
@@ -0,0 +1,304 @@
+package appgate
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/appgate/terraform-provider-appgate/client/v12/openapi"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAppgateSamlProvider() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAppgateSamlProviderRuleCreate,
+		Read:   resourceAppgateSamlProviderRuleRead,
+		Update: resourceAppgateSamlProviderRuleUpdate,
+		Delete: identityProviderDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Update: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: func() map[string]*schema.Schema {
+			s := identityProviderSchema()
+			s["type"].Default = identityProviderSaml
+
+			s["redirect_url"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			}
+			s["issuer"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			}
+			s["audience"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			}
+			s["provider_certificate"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			}
+			s["decryption_key"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			}
```

Comment:
sensitive?

Reply:
don't think its needed, since its included in the response body for `GET identity-provider/{id}`